sources/api/thar-be-settings/src/config.rs

```diff
@@ -76,11 +76,21 @@ pub async fn render_config_files(
             ));
         } else {
             match try_rendered {
-                Ok(rendered) => rendered_configs.push(RenderedConfigFile::new(
-                    &metadata.path,
-                    rendered,
-                    &metadata.mode,
-                )),
+                Ok(rendered) => {
+                    if metadata.skip_if_empty.is_some()
+                        && metadata.skip_if_empty.unwrap()
+                        && rendered.trim().is_empty()
```

Comment:
nit:

```suggestion
                    if metadata.skip_if_empty.unwrap_or(false)
                        && rendered.trim().is_empty()
```

feels a little more readable to me